### ISSUE TITLE
Help > Shortcuts button does nothing.

### ISSUE THREAD

**Issue by soakbot** [HaikuArchives/ArtPaint#21]
**Creator: Richard Sjusen Nyhus**
**URL: http://dev.osdrawer.net/issues/604**

_CAUTION: comments not extracted from issues. See OsDrawer or the WaybackMachine for issue comments._

**Comment by deep-1** [comment]
Is the URL mentioned is dead?

**Comment by richienyhus** [comment]
Yeah osdrawer has been dead since around 2013, HaikuArchives is kind of the spiritual successor to it.

Although I was the original bug reporter, I created the issue on behalf of a student in the 2012 Google Code-In contest, so I can't give you anymore details. I checked the original bug report with archive.org , but the original ticket didn't have any useful information.

**Comment by digib0y** [comment]
Clicking Help -> Shortcuts should open a pop-up and just give a list of shortcuts ?

**Comment by humdingerb** [comment]
"Shortcuts" used to be part of ArtPaint's accompanying documentation. "shortcuts.html" being one page of it. I'd suggest to import the complete docs into the HaikuArchives repo and thus make the "Help" and "Shortcuts" menu item work.
I found the "Documenatation" folder in an [older archive](http://pulkomandy.tk/~beosarchive/unsorted/svalka.freenet59.ru/BeOS/Graphics/Redactor/ArtPaint-2.0.zip). 

The docs can then be updated.

**Comment by richienyhus** [comment]
@humdingerb @digib0y The online version of the page is located here: http://haikuarchives.github.io/ArtPaint/shortcuts.html

That page comes from this branch of the ArtPaint repo:
https://github.com/HaikuArchives/ArtPaint/tree/gh-pages

**Comment by PeteCA** [comment]
Why are these files not in the master branch?  I  just built a "2.1.1" version of ArtPaint (http://goodeveCA.net/ArtPaint) that once again includes the (slightly updated) documentation (and some other fixes -- see the Pull Requests), but the docs really should be here and included in the recipe so they get to HaikuDepot as well.  I'll add the docs to my own github for a future PR.

**Comment by digib0y** [comment]
```
BPathFinder pathFinder;
	BStringList paths;
	BPath path;
	BEntry entry;

	status_t error = pathFinder.FindPaths(B_FIND_PATH_DOCUMENTATION_DIRECTORY,
		"packages/ArtPaint", paths);

	for (int i = 0; i < paths.CountStrings(); ++i) {
		if (error == B_OK && path.SetTo(paths.StringAt(i)) == B_OK
			&& path.Append("shortcuts.html") == B_OK) {
			entry = path.Path();
			entry_ref ref;
			entry.GetRef(&ref);
			be_roster->Launch(&ref);
		}
}
```

This should solve the issue ?

**Comment by humdingerb** [comment]
That would be one way to do it, though it's overly complicated.

OTOH, since ArtPaint is already in it's own Artpain folder under "apps" which contains also a "add-ons" folder, we could as well keep the documentation and tutorial etc. folders there, too.
That would entail finding the install location of ArtPaint at runtime. You can see how at e g. [QuickLaunch.cpp#L111](https://github.com/humdingerb/quicklaunch/blob/master/QuickLaunch.cpp#L111)

**Comment by PeteCA** [comment]
The only thing needed is to include the documentation folder in the artpaint one.

In the package I manually built (http://goodeveca.net/ArtPaint/artpaint-2.1.1-1-x86_gcc2.hpkg) I put the docs back there and made a (relative) link from the appropriate .../documentation directory to that (for user convenience -- not necessary for ArtPaint to see it).  Grab that and thake a look at the .packageinfo there if you liike.  (Beware -- I'll be replacing it with a newer version containing the addons later today.

**Comment by dsizzle** [comment]
This issue should be closed - Help->Shortcuts works fine in the 2.2.0 release

**Comment by humdingerb** [comment]
Indeed.
